aws-xray-auto-instrumentation-agent-runtime/src/main/java/com/amazonaws/xray/agent/handlers/downstream/HttpClientHandler.java

```diff
@@ -100,9 +102,12 @@ public void handleResponse(Event event) {
             return;
         }
 
+        // TODO: Refactor getSubsegment to getSubsegmentOptional and check for emptiness to avoid CMEs
```

Comment:
Curious what's `CME`? 

If it's possible, I'd generally recommend using `@Nullable` instead of `Optional` for new code since with a nullness checker like error prone it provides the same safety but has no runtime performance impact.

Reply:
CME = context missing exception, e.g. when you try to begin a subsegment without a segment present in the context. 

That's fair, `getSubsegment()` (which can raise a CME) and `getSubsegmentOptional()` (which cannot) are both methods on the recorder in the main Java SDK. So this change would probably have to be on the recorder itself.